mne/utils/check.py

```diff
@@ -1085,3 +1085,52 @@ def _to_rgb(*args, name='color', alpha=False):
         raise ValueError(
             f'Invalid RGB{"A" if alpha else ""} argument(s) for {name}: '
             f'{repr(args)}') from None
+
+
+def _check_pick_components(ica, picks):
+    """Validate picks for the given ICA and convert to an idx array."""
+    if ica.current_fit == "unfitted":
+        raise ValueError("The ICA must be fitted before selecting components.")
+
+    _validate_type(
+        picks,
+        (None, slice, range, tuple, list, np.ndarray, "int-like"),
+        "picks",
+    )
+
+    original_picks = picks
+    raise_ = False
+    if picks is None:
+        return np.arange(ica.n_components_)
+    elif isinstance(picks, (slice, range)):
+        if picks.start < 0 or ica.n_components_ < picks.stop:
+            raise_ = True
+        if picks.stop <= picks.start:
+            raise_ = True
+        picks = np.arange(ica.n_components_)[picks] if not raise_ else None
+    elif isinstance(picks, (tuple, list, np.ndarray)):
+        if isinstance(picks, np.ndarray):
+            if picks.ndim != 1:
+                raise_ = True
+            elif picks.size != np.unique(picks).size:
+                raise_ = True
+        picks = np.array(picks)
+        try:
+            if np.min(picks) < 0 or ica.n_components_ <= np.max(picks):
+                raise_ = True
+        except ValueError:  # zero-size array
+            raise_ = True
+    else:  # int
+        picks = _ensure_int(picks)
+        if picks < 0 or ica.n_components_ <= picks:
+            raise_ = True
+        picks = np.array([picks])
```

Comment:
These checks all look very similar to what is already in `_picks_to_idx`. Why can't we use that? You say it "picks on channel names" but this function should handle all these other cases you have here as well already

Reply:
Oh you can give it a number of channels instead of an Info! Then yes, replacing `_picks_to_idx(ica.info, picks)` with `_picks_to_idx(ica.n_components_, picks)` should work.